Add database reset suggestion [ci-skip] (#53583)
rails/rails: guides/source/active_record_migrations.md

migration can't be rolled back, `bin/rails db:reset` may not help you. To find
out more about dumping the schema see [Schema Dumping and You][] section.

> If you need an alternative to `db:reset` that explicitly runs all migrations,
> consider using the following command:
> 
> ```bash
> $ bin/rails db:drop db:create db:schema:dump db:migrate db:seed
> ```

Reviewer: ```suggestion
If you need an alternative to `db:reset` that explicitly runs all migrations,
consider using the `bin/rails db:migrate:reset` command. It is functionally
equivalent to `bin/rails db:drop db:create db:schema:dump db:migrate`.
You can follow that command with `bin/rails db:seed` if needed.
```

migration can't be rolled back, `bin/rails db:reset` may not help you. To find
out more about dumping the schema see [Schema Dumping and You][] section.

If you need an alternative to `db:reset` that explicitly runs all migrations,

> consider using the `bin/rails db:migrate:reset` command. It is functionally
> equivalent to `bin/rails db:drop db:create db:schema:dump db:migrate`. You can
> follow that command with `bin/rails db:seed` if needed.

Reviewer 2: ```suggestion
consider using the `bin/rails db:migrate:reset` command.
You can follow that command with `bin/rails db:seed` if needed.
```

> NOTE: `bin/rails db:reset` rebuilds the database using the current schema
> snapshot (`db/schema.rb` or `db/structure.sql`), ensuring consistency with the
> application's defined schema. On the other hand, `bin/rails db:migrate:reset`
> replays all migrations from the beginning, which can lead to schema drift if,
> for example, migrations have been altered, reordered, or removed.

Reviewer 2: ```suggestion
NOTE: `bin/rails db:reset` rebuilds the database using , ensuring consistency with the
application's defined schema. On the other hand, `bin/rails db:migrate:reset`
replays all migrations from the beginning, which can lead to schema drift if,
for example, migrations have been altered, reordered, or removed.
```

This is a bit wordy IMO.

> NOTE: `bin/rails db:reset` rebuilds the database using `db/schema.rb`, while `bin/rails db:migrate:reset`
replays all migrations from the beginning, which can lead to schema drift if migrations have been altered, reordered, or removed.

Author: Shortened.
